### PR TITLE
Sync repo refs (Please DNM: Do Not Merge)

### PR DESCRIPTION
**Pull request by jayvdb**
Fixes https://github.com/moremoban/moban/issues/284

### CODE REVIEW

**Comment by jayvdb** [comment]
ah crap, I just saw this has the wrong HEAD.

Anyways, I want to test it some more, and I also want to include a partial solution for https://github.com/moremoban/moban/issues/286 at the same time, so the syntax for repos with multiple remotes is considered and documented.

File: moban/repo.py [hunk 1]
```diff
@@ -24,11 +24,20 @@ def git_clone(requires):
...
-            if require.reference:
-                repo.git.checkout(require.reference)
-            elif require.branch:
-                repo.git.checkout(require.branch)
+            for remote in repo.remotes:
+                if require.git_url in remote.urls:
+                    break
+            else:
+                reporter.report_info_message("checking out new remote")
+                remote = Remote.create(repo, "current", require.git_url)
```
Comment (jayvdb) [inline moban/repo.py]: second time switching remotes causes:

'fatal: remote current already exists.'

Comment (jayvdb) [reply moban/repo.py]: Trying to create stable remote names for each remote is going to be difficult - and it should be done with enhancements in git-url-parse to allow matching similar remote names, e.g. discard unnecessary `.git` suffixes.

In order to fix this quickly, I propose doing a hash of the remote url.

**Comment by chfw** [comment]
I have verified and used this `branch`. I did not need to use remote but the branch is really switched. Even the branch is not local, moban is able to pull and switch to it. So, I am not sure on your use case.

**Comment by chfw** [comment]
@jayvdb , I managed to reproduce your problem once locally.

Please re-located your changes to https://github.com/moremoban/gitfs2. 

I am sorry for the inconvenience. Your changes in this PR will not go with moban v6 but with gitfs2.

**Pull request closed without merging**